tests/unit/amo/components/TestAutoSearchInput.js

```diff
@@ -316,7 +328,7 @@ describe(__filename, () => {
         root, query: 't'.repeat(SEARCH_TERM_MAX_LENGTH + 1),
       });
 
-      sinon.assert.notCalled(dispatchSpy);
+      sinon.assert.called(dispatchSpy);
```

Comment:
Since multiple actions could be dispatched here, you need to make sure it's the right one:

```js
sinon.assert.calledWith(dispatchSpy, autocompleteCancel());
```

You should also add this to make sure your `return` statement is in the right place:

```js
sinon.assert.callCount(dispatchSpy, 1);
```